### PR TITLE
Update documentation

### LINKED ISSUE #22: Installing should package
Since `should` package is registered as a `devDependency` and `peerDependency`, anyone that follows the instructions and isn't using `should` runs into an `UNMET PEER DEPENDENCY` warning followed by a `Cannot find module 'should'` error when running jscs.

The package is used in the `requireShouldAssertionExecution` rule, so I think we have two options: 

1. Move `should` from `devDependencies` to `dependencies`.
2. Add `should` to the [installation](https://github.com/seegno/jscs-config-seegno#installation) instructions on [README.md](https://github.com/seegno/jscs-config-seegno/blob/master/README.md).

### PR DESCRIPTION
**Pull request by ruiquelhas**
Resolves #22.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,13 @@
 # jscs-config-seegno
-
 Seegno-flavored JSCS config.
 
 ## Installation
 
 ```sh
-$ npm install jscs jscs-config-seegno --save-dev
+$ npm install jscs jscs-preset-seegno should --save-dev
```
Comment (ruimarinho) [inline README.md]: I would move the preset rename to another PR so we can rename the repo + package too.

Comment (ruiquelhas) [inline README.md]: Makes sense.

File: README.md [hunk 2]
```diff
@@ -31,3 +29,132 @@ and run the linter with:
 ```sh
 $ npm run lint
 ```
+
+## Custom rules
+The preset includes the following custom rules:
+
+### `disallowGeneratorsInDescribeFunctions`
+Disallows the usage of generators for the `describe` grouping primitive that features in testing frameworks such as `mocha`.
```
Comment (ruimarinho) [inline README.md]: Can you include a small paragraph on why this is critical to validate?

File: README.md [hunk 3]
```diff
@@ -31,3 +29,132 @@ and run the linter with:
...
+#### Invalid
+
+```js
+describe('foobar', function *() {
+  it('should work');
+});
+```
+
+### `disallowOnlyFilterInTestFunctions`
+Disallows the usage of `only` for grouping primitives or test cases in `mocha`.
```
Comment (ruimarinho) [inline README.md]: It'd be nice to mention why we don't want `.only` in tests.

File: README.md [hunk 4]
```diff
@@ -31,3 +29,132 @@ and run the linter with:
...
+  it('should work');
+});
+
+describe('foobar', function *() {
+  it.only('should work');
+});
+```
+
+### `requireShouldAssertionExecution`
+Disallows the usage of test expectation properties in favor of methods with libraries such as `should`.
```
Comment (ruimarinho) [inline README.md]: Also important to describe the side effect of not executing the assertion.

File: README.md [hunk 5]
```diff
@@ -31,3 +29,132 @@ and run the linter with:
...
+```
+
+#### Invalid
+
+```js
+true.should.be.true;
+```
+
+### `requireSqlTemplate`
+Disallows the usage of raw SQL templates with interpolation.
```
Comment (ruimarinho) [inline README.md]: Mention secure escaping provided by sql tag?

**Comment by ruimarinho** [comment]
Looks good minor small comments.

**Comment by ruiquelhas** [comment]
@ruimarinho I've added a small explanation for each rule.

**Comment by fixe** [comment]
LGTM.
